app/createTournament/createTournamentBusiness.js

```diff
@@ -0,0 +1,53 @@
+import * as common from '../../utils/common';
+import * as playerValidation from '../playersinscription/addPlayerBusiness';
+import {Player} from '../Player';
+import * as Math from '../../node_modules/mathjs/dist/math.min';
+
+export function validateNumberofPlayers(){
+
+var numberofPlayers = $('#tourneyPlayersList  >tbody >tr').length;
+  // 2^n in the future
+var validationResult={
+  success:true,
+  error:null
+};
+  if(numberofPlayers<8){
+    validationResult.success=false;
+    validationResult.error = {
+
+      lessthanrequired:true
+    };
+        addGhostPlayer(8-numberofPlayers);
+  }
+  if(numberofPlayers>8){
+    validationResult.success=false;
+    validationResult.error = {
+
+      morethanrequired:true
+    };
+  }
+  return validationResult;
+}
+
+function addGhostPlayer(number){
+  for (var i=0;i<number;i++){
+    playerValidation.addnewPlayer(new Player("Invited player number "+(i+1),addRandomRank()));
+  }
+}
+function addRandomRank(){
+
+  do{
+    var randomRankForInvitedPlayer=common.getRandomInt(1,100);
+
+  }while(common.isDuplicated(String(randomRankForInvitedPlayer)));
+return randomRankForInvitedPlayer;
```

Comment:
In some circumstances, this loop could never end. Don't rely on probabilities.

Furthermore, you don't want ghost players with high rankings. Instead, they should be at the end, so you could use max + 1.